react-context/flow.txt

```diff
@@ -0,0 +1,119 @@
+USER CAN SIGN UP.
+
+When a user visits the site, they will be taken to the log in page by default,
+which is represented by the sign-up hash.
+
+When user clicks 'register now' link or 'sign up' button, event listener fires,
+page is rerendered, hash changes, new welcome message, log in button changes to
+'register'. As user types in username and password, page rerenders with each
+keystroke.
+
+When the user clicks 'register' button a network request is sent to /api/auth/sign-up
+using POST method.
+
+Username and password become a stringified JSON object.
+
+When the server receives the request, it checks if the user fails to provide a
+username and/or password, a Client Error is thrown with 400 status code and error
+message.
+
+If both fields are supplied, the password is hashed, then the username and the
+password are inserted into the users table.
+
+The server will respond with a 201 status code, the userId, username, and
+createdAt information about the user to the client.
+
+React rerenders the page to show the log in form.
+
+This causes the hash of the location to change to sign-in
+
```

Comment:
These are actually backwards. The hash updates which triggers the event listener to update the current route in App.